Date: Sat, 28 Apr 2007 16:23:31 +1000
From: Herbert Xu <herbert.xu@redhat.com>
Subject: [XEN]: Set csum_start when setting csum_offset
Message-ID: <20070428062331.GA6461@gondor.apana.org.au>

On Fri, Apr 27, 2007 at 09:40:03PM -0300, Eduardo Pereira Habkost wrote:
> 
> On CVS, rpms/kernel-xen-2.6/devel, branch private-ehabkost-xen-3_0_5-branch.

Thanks.  It looks like somewhere along the line I managed to
lose an important line in the patch.


The new csum model requires csum_start to be set for csum_offset to
be meaningful.

Signed-off-by: Herbert Xu <herbert@gondor.apana.org.au>

Cheers,
-- 
Visit Openswan at http://www.openswan.org/
Email: Herbert Xu ~{PmV>HI~} <herbert@gondor.apana.org.au>
Home Page: http://gondor.apana.org.au/~herbert/
PGP Key: http://gondor.apana.org.au/~herbert/pubkey.txt
--
--- linux-2.6.20.i386/drivers/xen/core/skbuff.c	2007-04-28 15:30:16.000000000 +1000
+++ build-2.6.20.i386/drivers/xen/core/skbuff.c	2007-04-28 15:30:52.000000000 +1000
@@ -89,6 +89,7 @@
 	skb->h.raw = (unsigned char *)skb->nh.iph + 4*skb->nh.iph->ihl;
 	if (skb->h.raw >= skb->tail)
 		goto out;
+	skb->csum_start = skb->h.raw - skb->head;
 	switch (skb->nh.iph->protocol) {
 	case IPPROTO_TCP:
 		skb->csum_offset = offsetof(struct tcphdr, check);


